app/assets/javascripts/web_device_interface.ts

```diff
@@ -100,15 +98,39 @@ export class WebDeviceInterface extends DeviceInterface {
     return this.database.clearAllPayloads();
   }
 
-  getKeychainValue(): Promise<unknown> {
-    return this.bridge.getKeychainValue();
+  async getNamespacedKeychainValue() {
+    const keychain = await this.getRawKeychainValue();
+    if (!keychain) {
+      return;
+    }
+    return keychain[this.namespace!.identifier];
+  }
+
+  async setNamespacedKeychainValue(value: any) {
+    let keychain = await this.getRawKeychainValue();
+    if (!keychain) {
+      keychain = {};
+    }
+    localStorage.setItem(this.bridge.keychainStorageKey, JSON.stringify({
+      ...keychain,
+      [this.namespace!.identifier]: value,
+    }));
```

Comment:
Should be `this.bridge.setKeychainValue({ ...keychain, [this.namespace!.identifier]: value })` instead. On desktop we use the OS keychain instead of localStorage, hence the extraction of the keychain methods to a common `Bridge` interface